exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetComparisonPredicate.java

```diff
@@ -83,23 +84,26 @@ private ParquetComparisonPredicate(
    * where Column1 and Column2 are from same parquet table.
    */
   @Override
-  public boolean canDrop(RangeExprEvaluator<C> evaluator) {
+  public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
     Statistics<C> leftStat = left.accept(evaluator, null);
-    if (isNullOrEmpty(leftStat)) {
-      return false;
+    if (isNullOrEmpty(leftStat) || !leftStat.hasNonNullValue()) {
+      return RowsMatch.SOME;
     }
-
     Statistics<C> rightStat = right.accept(evaluator, null);
-    if (isNullOrEmpty(rightStat)) {
-      return false;
+    if (isNullOrEmpty(rightStat) || !rightStat.hasNonNullValue()) {
+      return RowsMatch.SOME;
     }
-
-    // if either side is ALL null, = is evaluated to UNKNOWN -> canDrop
     if (isAllNulls(leftStat, evaluator.getRowCount()) || isAllNulls(rightStat, evaluator.getRowCount())) {
-      return true;
+        return RowsMatch.NONE;
     }
+    return predicate.apply(leftStat, rightStat);
+  }
 
-    return (leftStat.hasNonNullValue() && rightStat.hasNonNullValue()) && predicate.test(leftStat, rightStat);
+  /**
+   * If one rowgroup contains some null values, change the RowsMatch.ALL into RowsMatch.SOME (null values should be discarded by filter)
+   */
+  static RowsMatch checkNull(Statistics leftStat, Statistics rightStat) {
```

Comment:
Please see my other comment for `checkNull`.

Reply:
Done